pymc3/tests/test_distributions_random.py

```diff
@@ -492,43 +421,316 @@ class TestMoyal(BaseTestCases.BaseTestCase):
+    arguments all the parameters needed by the checks and the list of checks to be ran.
 
-    def test_probability_vector_shape(self):
-        """Check that if a 2d array of probabilities are passed to categorical correct shape is returned"""
-        p = np.ones((10, 5))
-        assert pm.Categorical.dist(p=p).random().shape == (10,)
-        assert pm.Categorical.dist(p=p).random(size=4).shape == (4, 10)
-        p = np.ones((3, 7, 5))
-        assert pm.Categorical.dist(p=p).random().shape == (3, 7)
-        assert pm.Categorical.dist(p=p).random(size=4).shape == (4, 3, 7)
+    Some testing needs are currently taken care of by the aesara library, so we don't need to run all checks for all the
+    distributions
+    """
 
+    def _run_distribution_checks(
+        self, params: Dict[str, Any], tests_to_run: List[Callable]
+    ) -> None:
+        params["pymc_dist_output"] = params["pymc_dist"].dist(
+            **params["pymc_dist_params"],
+            size=params.get("size", 15),
+            rng=aesara.shared(self.get_random_state()),
+        )
+        if "expected_dist" in params:
+            params["expected_dist_outcome"] = params["expected_dist"](
+                **params["expected_dist_params"], size=params.get("size", 15)
+            )
+        for test in tests_to_run:
+            test(**params)
 
-@pytest.mark.skip(reason="This test is covered by Aesara")
-class TestDirichlet(SeededTest):
-    @pytest.mark.parametrize(
-        "shape, size",
-        [
-            ((2), (1)),
-            ((2), (2)),
-            ((2, 2), (2, 100)),
-            ((3, 4), (3, 4)),
-            ((3, 4), (3, 4, 100)),
-            ((3, 4), (100)),
-            ((3, 4), (1)),
-        ],
-    )
-    def test_dirichlet_random_shape(self, shape, size):
-        out_shape = to_tuple(size) + to_tuple(shape)
-        assert pm.Dirichlet.dist(a=np.ones(shape)).random(size=size).shape == out_shape
+    def _check_pymc_dist_matches_expected(
+        self,
+        pymc_dist_output: TensorVariable,
+        expected_dist_outcome: np.ndarray,
+        decimal: int = 6,
+        **kwargs,
+    ) -> None:
+        assert_array_almost_equal(pymc_dist_output.eval(), expected_dist_outcome, decimal=decimal)
+
+    def _check_pymc_params_match_rv_op(
+        self,
+        pymc_dist_output: TensorVariable,
+        expected_rv_op_params: Dict[str, Any],
+        decimal: int = 6,
+        **kwargs,
+    ) -> None:
+        aesera_dist_inputs = self._get_inputs_from_apply_node_outputs(pymc_dist_output)[3:]
+        assert len(expected_rv_op_params) == len(aesera_dist_inputs)
+        for (expected_name, expected_value), actual_variable in zip(
+            expected_rv_op_params.items(), aesera_dist_inputs
+        ):
+            assert_almost_equal(expected_value, actual_variable.eval(), decimal=decimal)
+
+    def _get_inputs_from_apply_node_outputs(self, outputs: TensorVariable) -> List[Any]:
```

Comment:
This might not justify the separate function, unless it starts to be used in more than one type of test

Reply:
Yeah, I tend to prefer using a lot of small functions rather than a few big ones in which the core logic is buried in a lot of code that is only massaging data. I can remove it